src/main/java/seedu/address/model/sale/Sale.java

```diff
@@ -90,28 +100,29 @@ public boolean equals(Object other) {
 
         Sale otherSale = (Sale) other;
         return otherSale.getItemName().equals(getItemName())
-                && otherSale.getQuantity().equals(getQuantity())
+                && otherSale.getDatetimeOfPurchase().equals(getDatetimeOfPurchase())
                 && otherSale.getUnitPrice().equals(getUnitPrice())
+                && otherSale.getQuantity().equals(getQuantity())
                 && otherSale.getTags().equals(getTags());
     }
 
     @Override
     public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
-        return Objects.hash(itemName, quantity, unitPrice);
+        return Objects.hash(itemName, datetimeOfPurchase, quantity, unitPrice);
     }
 
     @Override
     public String toString() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getItemName())
-                .append(" (Quantity: ")
+                .append(" (Date of Purchase: ")
+                .append(getDatetimeOfPurchase())
```

Comment:
Should we format the datetime so that it is consistent with the Meeting `toString()` datetime format?

Reply:
Fixed in latest commit!